### PR TITLE
Feat: Migrate all compose dependencies to convention plugin

### PR DESCRIPTION
**Pull request by misshannah**
# Scope

_Please make sure to read the [Contribution Guidelines](https://github.com/droidconKE/droidconKE2023Android/blob/main/CONTRIBUTING.md)
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [x] I have followed the coding conventions
- [x] I have added/updated necessary tests
- [x] I have tested the changes added on a physical device
- [x] I have run `./codeAnalysis.sh` on linux/unix or `codeAnalysys.bat` on windows to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
Fixes part of issue33

## Other testing QA Notes
_What have you tested specifically and what possible impacts/areas there are that may need retesting
by others._

Please add a screenshot (if necessary)

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -89,6 +91,7 @@ firebase-performance = { module = "com.google.firebase:firebase-perf-ktx" }
 firebase-config = { module = "com.google.firebase:firebase-config-ktx" }
 chucker-debug = { module = "com.github.chuckerteam.chucker:library", version.ref = "chucker" }
 chucker-release = { module = "com.github.chuckerteam.chucker:library-no-op", version.ref = "chucker" }
+androidx-compose-material3-windowSizeClass = { group = "androidx.compose.material3", name = "material3-window-size-class" }
```
Comment (wangerekaharun) [inline gradle/libs.versions.toml:94]: We're not using WindoSizeClass in the proejct

**Comment by wangerekaharun** [comment]
@misshannah some Ui tests are failing

File: gradle/libs.versions.toml (hunk 1 above)
Comment (misshannah) [reply gradle/libs.versions.toml:94]: I'll remove it.

File: build-logic/convention/src/main/kotlin/com/android254/AndroidCompose.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    commonExtension.apply {
+        buildFeatures {
+            compose = true
+        }
+
+        composeOptions {
+            kotlinCompilerExtensionVersion = libs.findVersion("composecompiler").get().toString()
+        }
+
+        dependencies {
```
[the remark this answers is not in the document]
Comment (misshannah) [reply build-logic/convention/src/main/kotlin/com/android254/AndroidCompose.kt:40]: Oh oh, they don't seem to work with this new setup. I can try to troubleshoot.

[the remark this answers is not in the document]
Comment (wangerekaharun) [reply build-logic/convention/src/main/kotlin/com/android254/AndroidCompose.kt:40]: You can add this two libraries to fix this:
- androidx.customview:customview:1.2.0-alpha01
- androidx.customview:customview-poolingcontainer:1.0.0-alpha01

You can refer here: https://issuetracker.google.com/issues/227767363

[the remark this answers is not in the document]
Comment (misshannah) [reply build-logic/convention/src/main/kotlin/com/android254/AndroidCompose.kt:40]: They are already in the list of libraries. I now think it's something to do with my Android Studio. I'll update you when I make progress.

**Review by wangerekaharun — APPROVED** [review]

**Review by chepsi — APPROVED** [review]

**Pull request merged**